### DISCUSSION TITLE
[messaging/unregistered] You must be registered for remote messages before calling getToken, see messaging().registerDeviceForRemoteMessages().

### DISCUSSION THREAD

**Discussion by euharrison** [invertase/react-native-firebase#4646] [Q&A]
Hi,

I'm receiving this error message on Sentry
> [messaging/unregistered] You must be registered for remote messages before calling getToken, see messaging().registerDeviceForRemoteMessages().

But I don't have a `firebase.json` in my project neither disabled the auto-registration option.

And if I manually enable the registration in my app by `messaging().registerDeviceForRemoteMessages()`, I receive this warn
> Usage of "messaging().registerDeviceForRemoteMessages()" is not required. You only need to register if auto-registration is disabled in your 'firebase.json' configuration file via the 'messaging_ios_auto_register_for_remote_messages' property.

Even if I check for registration with
```js
if (!messaging().isDeviceRegisteredForRemoteMessages) {
  await messaging().registerDeviceForRemoteMessages();
}
```

Does anyone has any tips how to handle this? Or at least just remove this warning?

**Comment by mikehardy** [comment]
I am not certain on this, but have you tried creating an empty firebase.json? I have a vague recollection that fixed something else, and may result in default initialization (and thus no warnings) as expected

**Comment by euharrison** [comment]
Thanks for answer!

I will try this option with the next release and keep tracking if it'll be resolved

**Comment by euharrison** [comment]
No luck, the problem persist :(

I created the file `firebase.json` with this content
```json
{
  "react-native": {
    "messaging_ios_auto_register_for_remote_messages": true
  }
}
```

Apparently this happen only on iOS

Does someone has another tip?

**Reply by mikehardy** [reply]
I suggest opening an issue and following the template since this is not really a discussion, it is a problem. First make sure you can reproduce it with current versions in a clean demo, I provide a script to do so here: https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh
